core/src/main/java/tc/oc/pgm/action/ActionModule.java

```diff
@@ -48,7 +49,7 @@ public static class Factory implements MapModuleFactory<ActionModule> {
 
     @Override
     public Collection<Class<? extends MapModule<?>>> getWeakDependencies() {
-      return ImmutableList.of(VariablesModule.class);
+      return ImmutableList.of(VariablesModule.class, StructureModule.class);
```

Comment:
You should not need this dependency. if you just use references you don't need to force load order